nucleus/modelci/unit_test_metric.py

```diff
@@ -0,0 +1,14 @@
+from nucleus.pydantic_base import ImmutableModel
+
+from .constants import ThresholdComparison
+
+
+class UnitTestMetric(ImmutableModel):
```

Comment:
When should we use ImmutableModel and when should we use Dataclass?

Reply:
I'd suggest everything over the wire (`data_transfer_objects`) should be a `ImmutableModel` since we benefit from `pydantic` validation. Dataclasses are then used as the domain objects that can be created from the `data_transfer_objects`.